chaincode/src/allowances/useAllowances.ts

```diff
@@ -98,8 +98,10 @@ export async function useAllowances(
 
     await putChainObject(ctx, newClaim);
 
-    // Update the allowance in the context
-    await putChainObject(ctx, tokenAllowance);
+    // Update the allowance in the context if quantity is finite
+    if (tokenAllowance.quantity.isFinite()) {
+      await putChainObject(ctx, tokenAllowance);
+    }
```

Comment:
this is actually an inconsistency, in two ways:
1. We update `tokenAllowance` object before, and now we don't save changes. If you change some object that was read, while reading the code, you expect it to be saved, and the final if statement might be a surprise.
2. With no update on allowance, all `*Spent` fields remain at `0`, which also might be misleading. And I know we don't want to update it (performance). So we need to have a better default (`undefined`?). This way when infinite allowance is created it would get `undefined` for all `*Spent` fields and we won't update it at all.